### DISCUSSION TITLE
Does this support Postgres Certificate Authentication?

### DISCUSSION THREAD

**Discussion by andy-g** [jeroenrinzema/psql-wire#65] [Q&A] [answered]
I'm looking to build a tool that'll allow users to connect to a set of backend Postgres DB's, but would prefer to manage access in this proxy, rather than create users on each backend postgres instance individually. Ideally I'd like to configure postgres DB's with a shared user as backend sources, and then users will be able to authenticate to this proxy, which will pass their queries to the backend server if they have the required access. I'd ideally want to log their queries at the same time.

It looks like psql-wire would be the perfect foundation for this, but I'd ideally like to avoid managing user credentials, so was wondering if psql-wire supports Postgres Certificate Authentication (https://www.postgresql.org/docs/current/auth-cert.html).

Also, does anyone have a suggestion of a tools that already caters for the needs I've outlined above? I started looking at https://cloudproud.nl/ (found the link on your GitHub page), but ran into some bugs, so not sure how ready it is for use in production, and wasn't able to find a GitHub repo for their service.

**Comment by jeroenrinzema** [comment, marked as answer]
Hi Andy, thanks for reaching out! The wire protocol does not yet support PostgreSQL certificate authentication. This authentication flow has to be included in the project. The PSQL wire protocol would be able to allow users to connect to your solution and access all their data using SQL queries. The wire protocol project does not support the parsing of the incoming queries and the validating of user access policies. This is exactly what inspired us to create SeamKit. SeamKit is currently in open beta and allows users to connect multiple sources as a single virtual database. We would love to hear your feedback on the kit and help you resolve the issues you are facing. Feel free to reach out to me on our public slack channel to discuss this further.
